src/backgroundGCP/DataIngestionGCP/Functions/GetCopilotSeatsByDate.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Google.Cloud.Firestore;
+using Google.Cloud.Functions.Framework;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CopilotDashboard.DataIngestion.Functions;
+
+public class GetCopilotSeatsByDate : IHttpFunction
+{
+    private readonly ILogger<GetCopilotSeatsByDate> _logger;
+    private readonly FirestoreDb _firestoreDb;
+    private readonly JsonSerializerOptions jsonSerializerOptions = new()
+    {
+        Converters = { new FirestoreTimestampConverter() }
+    };
+
+    public GetCopilotSeatsByDate(
+    ILogger<GetCopilotSeatsByDate> logger,
+    FirestoreDb firestoreDb)
+    {
+        _logger = logger;
+        _firestoreDb = firestoreDb;
+    }
+
+    public async Task HandleAsync(HttpContext context)
+    {
+        try
+        {
+            string apiKey = context.Request.Query["apiKey"];
```

Comment:
I think it's better to pass it as a header for more security

Reply:
Added header support. Per discussion in the SU, we will also maintain query string parameter support.